### DISCUSSION TITLE
Difficulties with typescript/next-auth module importing

### DISCUSSION THREAD

**Discussion by tallowen** [vikejs/vike#749] [Discussion]
I've started a new project using the following steps

```
 npm init vite-plugin-ssr@latest # using vue / typescript
 cd test-import-issue/
 npm install
 npm install next-auth # installs next auth v4
```

In `server/index.ts` I added the following lines as a proof of concept:
```
import CredentialsProvider from "next-auth/providers/credentials"
console.log(CredentialsProvider);
CredentialsProvider({})
```

When I run `npm run dev` I get the following output:
```
➜  test-import-issue git:(master) ✗ npm run dev

> dev
> npm run server


> server
> ts-node ./server/index.ts

{ default: [Function: Credentials] }
TypeError: CredentialsProvider is not a function
    at file:///Users/owencoutts/dev/test-import-issue/server/index.ts:9:1
    at ModuleJob.run (node:internal/modules/esm/module_job:194:25)
```

I tried a couple of things to no avail:
- Because this seems like a module loading issue, add it to the noExternal vite config [as described in the docs](https://vite-plugin-ssr.com/invalid-esm#solution)
- I  tried using [vite-ssr-next-auth)](https://github.com/s-kris/vite-ssr-next-auth) as a starting point. Although I could get everything running using next-auth version 3,  I noticed the same issues when I upgraded to version 4.

Apologies in advanced for being new to vite and not being sure how to debug these module issues!

**Comment by brillout** [comment]
I don't know, but it seems like a small error on the user land you overlooked. Keep me updated and let me know if you do believe the error to be related to Vite.

**Comment by schaschko** [comment]
@tallowen 

Use `CredentialsProvider.default()`, s. this [issue](https://github.com/nextauthjs/next-auth/discussions/3942#discussioncomment-2793945).
